### ISSUE TITLE
Unhelpful compile error for compute shader

### ISSUE THREAD

**Issue by HalfVoxel** [dfranx/SHADERed#188]
Hi

Thanks for making this, it looks very useful!

I tried adding a compute shader which compiles with shaderc, but I get a very unhelpful message "Failed to compile the compute shader" and no other errors. I don't really have any idea of how to further debug it.

This is the shader, a stupid simple blur shader:

```glsl
#version 450

layout(local_size_x = 32, local_size_y = 32, local_size_z = 1) in;
layout (set=0, binding=0, rgba8) uniform restrict image2D myStorageImage;

void main() {
    ivec2 px = ivec2(gl_WorkGroupID.xy*uvec2(32, 32) + gl_LocalInvocationID.xy);

    vec4 res = vec4(0.0, 0.0, 0.0, 0.0);

    for (int dx = -8; dx <= 8; dx++) {
        for (int dy = -8; dy <= 8; dy++) {
            res += imageLoad(myStorageImage, px + ivec2(dx, dy));
        }
    }
    res /= 17*17;
    
    imageStore(myStorageImage, px, res);
}
```

![image](https://user-images.githubusercontent.com/1144597/111340927-57f87e80-8679-11eb-9f5f-236c0f060f36.png)

![image](https://user-images.githubusercontent.com/1144597/111340947-5b8c0580-8679-11eb-9c19-2e6dc0dde1d7.png)

**Comment by dfranx** [comment]
Hey!

Sorry for the inconvenience, this is a weird quirk in SHADERed - `glslang` (compiler) didn't report any error (since it's a valid GLSL code), but when uploading that code to GPU an error is reported (and I don't print those in the "Output" window).

`layout(set=...)` is supported in Vulkan's version of GLSL (and SHADERed uses OpenGL), as far as I am aware. 
You could add this code below the `#version 450` line:
```glsl
#extension GL_KHR_vulkan_glsl : enable
```
I tested your code with the `#extension` line and it compiles fine.

Or you could change the extension to `.vk` (so that SHADERed's compiler thinks of it as Vulkan GLSL).. Though, adding `#extension` might be a better solution since Vulkan GLSL is not truly supported.

**Comment by HalfVoxel** [comment]
Thanks!

I still can't figure out how to actually use the compute shader, but that's unrelated to this issue.

Leaving this issue open since it is still an unhelpful message :)

**Comment by dfranx** [comment]
This should now be fixed and included in the next release of SHADERed (v1.5.1).
![BetterErrorMessages](https://user-images.githubusercontent.com/30801537/112398979-a35bfe00-8d05-11eb-839a-0f640ee0591f.png)

By the way, by "can't figure out how to use the compute shader", do you mean in SHADERed or generally? If it's due to SHADERed, feel free to email me!

I think I can close this issue now as it has been resolved.
